packages/react-native-web/src/exports/Image/index.js

```diff
@@ -332,14 +343,67 @@ const Image: React.AbstractComponent<
   );
 });
 
-Image.displayName = 'Image';
+BaseImage.displayName = 'Image';
+
+/**
+ * This component handles specifically loading an image source with headers
+ * default source is never loaded using headers
+ */
+const ImageWithHeaders: ImageComponent = React.forwardRef((props, ref) => {
+  // $FlowIgnore
+  const nextSource: ImageSource = props.source;
+  const [blobUri, setBlobUri] = React.useState('');
+  const request = React.useRef<LoadRequest>({
+    cancel: () => {},
+    source: { uri: '', headers: {} },
+    promise: Promise.resolve('')
+  });
+
+  const { onError, onLoadStart, onLoadEnd } = props;
+
+  React.useEffect(() => {
+    if (!hasSourceDiff(nextSource, request.current.source)) return;
+
+    // When source changes we want to clean up any old/running requests
+    request.current.cancel();
+
+    if (onLoadStart) onLoadStart();
+
+    request.current = ImageLoader.loadWithHeaders(nextSource);
+
+    request.current.promise
+      .then((uri) => setBlobUri(uri))
+      .catch(() =>
+        raiseOnErrorEvent(request.current.source.uri, { onError, onLoadEnd })
+      );
+  }, [nextSource, onLoadStart, onError, onLoadEnd]);
+
+  // Cancel any request on unmount
+  React.useEffect(() => request.current.cancel, []);
+
+  const propsToPass = {
+    ...props,
+    // Omit `onLoadStart` because we trigger it in the current scope
```

Comment:
I think it's more like: `ImageWithHeaders` already calls `onLoadStart` when it starts loading the image, so we don't want the `BaseImage` to trigger that function a second time

Reply:
Now that I look at it I see it's confusing - it's like Alex said - loading starts inside `ImageWithHeaders`, to prevent `BaseImage` to raise `onLoadStart` a second time we filter it out

Reply:
Ok cool I think we are all saying the same thing - `BaseImage` will not use an `onLoadStart` callback when we have headers.